### PR TITLE
Support for 16 bit short and ushort

### PR DESCRIPTION
**Pull request by shehzan10**
Ref #357

### CODE REVIEW

**Comment by shehzan10** [comment]
@pavanky this can be merged. The OSX failure is for linear algebra functions, The windows pull was successful but for did not update for some reason.

File: src/api/c/implicit.cpp [hunk 1]
```diff
@@ -47,6 +47,12 @@ af_dtype implicit(const af_dtype lty, const af_dtype rty)
     if ((lty == s32) ||
         (rty == s32)) return s32;
 
+    if ((lty == u16) ||
+        (rty == u16)) return u16;
+
+    if ((lty == s16) ||
+        (rty == s16)) return s16;
+
```
Comment (pavanky) [inline src/api/c/implicit.cpp]: :+1:

Comment (pavanky) [inline src/api/c/implicit.cpp]: I did not think of this until I actually saw the change. Can you look at src/api/cpp and ensure there is no implicit.cpp in there as well ?

Comment (pavanky) [inline src/api/c/implicit.cpp]: Nevermind the C++ version is good.

Comment (shehzan10) [inline src/api/c/implicit.cpp]: No its not there. I was quite thorough in the changes.

**Comment by pavanky** [comment]
good work @shehzan10

**Pull request merged**